File: catalog/backends/java-spring/patterns/FeatureFlagStringVariant.java
package com.example.service;

import dev.openfeature.sdk.*;

// Pattern: Feature Flags — string variant for A/B branching
// Adapt: replace "ui-theme" with your flag key; variant values must match Datadog flag config
public class FeatureFlagStringVariant {

    private static Client getClient() {
        return OpenFeatureAPI.getInstance().getClient(); // provider initialized in FeatureFlagBooleanGate
    }

    public String getUiVariant(String userId) {
        EvaluationContext ctx = new MutableContext(userId);
        String variant = getClient().getStringValue("ui-theme", "control", ctx);
        return switch (variant) {
            case "treatment-a" -> "new_design_a";
            case "treatment-b" -> "new_design_b";
            default -> "control_design";
        };
    }
}

File: catalog/backends/dotnet-aspnetcore/patterns/FeatureFlagStringVariant.cs
using OpenFeature;
using OpenFeature.Model;

// Pattern: Feature Flags — string variant for A/B branching
// Adapt: replace "ui-theme" with your flag key; variant values must match Datadog flag config
public class FeatureFlagStringVariant
{
    private static readonly FeatureClient _client = Api.Instance.GetClient("service"); // provider initialized in FeatureFlagBooleanGate

    private static readonly Dictionary<string, string> _variants = new()
    {
        ["treatment-a"] = "new_design_a",
        ["treatment-b"] = "new_design_b",
    };

    public async Task<string> GetUiVariantAsync(string userId)
    {
        var ctx = EvaluationContext.Builder().SetTargetingKey(userId).Build();
        var variant = await _client.GetStringValueAsync("ui-theme", "control", ctx);
        return _variants.GetValueOrDefault(variant, "control_design");
    }
}
